src/Sylius/Bundle/CoreBundle/ShopFixtures/Symfony/Messenger/MessengerCommandBus.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\ShopFixtures\Symfony\Messenger;
+
+use Sylius\Bundle\CoreBundle\ShopFixtures\Command\CommandInterface;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\HandleTrait;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class MessengerCommandBus implements CommandBusInterface
+{
+    use HandleTrait;
+
+    public function __construct(MessageBusInterface $commandBus)
+    {
+        $this->messageBus = $commandBus;
+    }
```

Comment:
To avoid infinite loop in dependency injection with Factory::findOrCreate everywhere, I used the event system in previous implementation. I think it's better with CQRS and this is not so coupled to Messenger, cause this is just some class with invoke method.